src/preamble.js

```diff
@@ -653,7 +653,13 @@ if (Module['locateFile']) {
 #if EXPORT_ES6 && USE_ES6_IMPORT_META && !SINGLE_FILE // in single-file mode, repeating WASM_BINARY_FILE would emit the contents again
 } else {
   // Use bundler-friendly `new URL(..., import.meta.url)` pattern; works in browsers too.
-  wasmBinaryFile = new URL('{{{ WASM_BINARY_FILE }}}', import.meta.url).toString();
+  wasmBinaryFile = new URL('{{{ WASM_BINARY_FILE }}}', import.meta.url);
+#if ENVIRONMENT_MAY_BE_NODE
+  if (ENVIRONMENT_IS_NODE) {
+    wasmBinaryFile = nodeUrl.fileURLToPath(wasmBinaryFile);
```

Comment:
I think this step shouldn't be necessary.

Node.js `fs.readFile` can read files by their `file://` URLs in all versions that have ESM support (in fact, construction of URLs relative to `import.meta.url` was one of the main reasons for adding URL support to `fs`).

Reply:
Good point! Should be fixed with commit cff5945db24ffc560421f28759efc2993f931925, I only needed to convert the `file://` string back into URL to allow it to pass to `fs.readFileSync`, but that's all.